Add purchases for CUBE study labs

mrgnr opened the pull request:
## Done

The "Prepare" button now displays the purchase modal if study labs have not yet been purchased. Otherwise, it links to the corresponding study lab.

Depends on https://github.com/canonical-web-and-design/ubuntu.com/pull/10689

## QA

- Run the server with `dotrun`
- Go to http://localhost:8001/cube/microcerts?test_backend=true
- Click on the "Prepare" button and complete the purchase process
- Once the page reloads, click on the "Prepare" button to go to the study lab
- Go to https://qa.cube.ubuntu.com/dashboard and unenroll yourself from the "CUBE 2020 Study Labs" course (click on the gear icon and select "Unenroll")
- Go to http://localhost:8001/cube/study?test_backend=true
- Click on the "Purchase study labs access" button and complete the purchase process
- Once the page reloads, click on the "Access study labs" button to go to the study labs

## Issue / Card

Fixes https://github.com/canonical-web-and-design/commercial-squad/issues/348


## Screenshots

### /cube/microcerts

![cube-prepare](https://user-images.githubusercontent.com/995051/139833686-75792edd-847c-4fa8-bcc0-f0382140e92b.png)


### /cube/study
![cube-study](https://user-images.githubusercontent.com/995051/139829261-b1894f39-9a8a-4a0c-8c73-d54224a8dc19.png)

wgx commented:
@mrgnr 

>Given the CUBE site's current structure, the button on https://ubuntu.com/cube/microcerts SHOULD say "Prepare", as it leads to the free study material. The button on https://ubuntu.com/cube/study should say "Purchase", because that's where customers can decide to buy the labs. There is copy on that page explaining what they are for.

This change (4) from Lex sounds ideal, please do implement this too.

mrgnr commented:
@wgx

> 1. The "M" icon is filled - yet the one I have passed has a different icon, is this a bug:

I'll look into this.

> 2. Upon a purchase, the page reloads but the newly purchased item still says "purchase" - known issue?

It seems that demos.haus uses the production edX instance rather than QA, so I think this PR needs to be tested locally.

> 3. We need some copy for this section:

Updated (I'd accidentally removed what what there before :laughing: )

> This change (4) from Lex sounds ideal, please do implement this too.

Updated.

carkod commented:
> @wgx
> 
> > 1. The "M" icon is filled - yet the one I have passed has a different icon, is this a bug:
> 
> I'll look into this.
> 
> > 1. Upon a purchase, the page reloads but the newly purchased item still says "purchase" - known issue?
> 
> It seems that demos.haus uses the production edX instance rather than QA, so I think this PR needs to be tested locally.
> 
> > 1. We need some copy for this section:
> 
> Updated (I'd accidentally removed what what there before laughing )
> 
> > This change (4) from Lex sounds ideal, please do implement this too.
> 
> Updated.

Point 2 looks like a backend issue, but I can't reproduce it. I just gave it a try with this PR and it does actually change the buttons, maybe it's been fixed already?

carkod commented:
> @wgx
> 
> > 1. The "M" icon is filled - yet the one I have passed has a different icon, is this a bug:
> 
> I'll look into this.
> 
> > 2. Upon a purchase, the page reloads but the newly purchased item still says "purchase" - known issue?
> 
> It seems that demos.haus uses the production edX instance rather than QA, so I think this PR needs to be tested locally.
> 
> > 3. We need some copy for this section:
> 
> Updated (I'd accidentally removed what what there before laughing )
> 
> > 4. This change (4) from Lex sounds ideal, please do implement this too.
> 
> Updated.

Issues 1 and 2 potentially will be fixed in https://github.com/canonical-web-and-design/ubuntu.com/pull/10788

mrgnr commented on static/js/src/cube/hooks/useMicrocertsData.tsx:
```diff
@@ -0,0 +1,57 @@
+  const [studyLabs, setStudyLabs] = useState<Record<string, unknown>>({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState("");
+
```
Thanks for the suggestion! Done.
